core/math/camera_matrix.cpp

```diff
@@ -123,6 +123,60 @@ void CameraMatrix::set_perspective(real_t p_fovy_degrees, real_t p_aspect, real_
 	*this = *this * cm;
 }
 
+void CameraMatrix::set_oblique(real_t p_fovy_degrees, real_t p_aspect, real_t p_z_near, real_t p_z_far, Quat p_oblique_plane, bool p_flip_fov) {
+	if (p_flip_fov) {
+		p_fovy_degrees = get_fovy(p_fovy_degrees, 1.0 / p_aspect);
+	}
+
+	real_t sine, cotangent, deltaZ;
+	real_t radians = p_fovy_degrees / 2.0 * Math_PI / 180.0;
+
+	deltaZ = p_z_far - p_z_near;
+	sine = Math::sin(radians);
+
+	if ((deltaZ == 0) || (sine == 0) || (p_aspect == 0)) {
+		return;
+	}
+	cotangent = Math::cos(radians) / sine;
+
+	set_identity();
+
+	matrix[0][0] = cotangent / p_aspect;
+	matrix[1][1] = cotangent;
+	matrix[2][2] = -(p_z_far + p_z_near) / deltaZ;
+	matrix[2][3] = -1;
+	matrix[3][2] = -2 * p_z_near * p_z_far / deltaZ;
+	matrix[3][3] = 0;
+
+	// Here goes oblique magic!
+	// Eric Lengyel Solution: http://terathon.com/code/oblique.html
+	// i < 0 ? -1 : (i > 0 ? +1 : 0) = sign()
+	Quat q;
+
+	/* clang-format off */
+	q.x = ((
+		p_oblique_plane.x < 0 ? -1 :
+		p_oblique_plane.x > 0 ? +1 : 0
+		) + matrix[2][0]
+	) / matrix[0][0];
+
+	q.y = ((
+		p_oblique_plane.y < 0 ? -1 :
+		p_oblique_plane.y > 0 ? +1 : 0
+		) + matrix[2][1]
+	) / matrix[1][1];
+	/* clang-format on */
```

Comment:
I think you can use the `SGN` macro to get -1, +1, and 0. The `clang-format` comments can then be removed hopefully.

Reply:
I don't think SGN will work here... should I opt for intermediary variables?
`#define SGN(m_v) (((m_v) < 0) ? (-1.0) : (+1.0))`

Reply:
Ah, please don't. That'll break compatibility I think.

But the `clang-format` tweak still seems awkward to me, maybe something can be done to avoid that...

Reply:
sorry, I wasn't suggesting changing the macro, just including it to show it doesn't have the necessary 'zero if zero' branch.

This is the adapted algorithm used in my PR:
http://terathon.com/code/oblique.html

Specifically, he uses an inline function:
![image](https://user-images.githubusercontent.com/40372043/148779041-6732cf42-091d-4d32-abd9-8d0470f1b012.png)

which you can see does greatly simplify the expression:
![image](https://user-images.githubusercontent.com/40372043/148779087-b0c81d78-ad65-4096-b625-2716aecf3a3c.png)

Reply:
In master, it's:

```cpp
#ifndef SIGN
#define SIGN(m_v) (((m_v) == 0) ? (0.0) : (((m_v) < 0) ? (-1.0) : (+1.0)))
#endif
```

We could backport `SIGN` with the behavior of `0.0` returning `0.0` and keep `SGN` the way it is.

Reply:
That seems like a reasonable option, especially given that according to Wikipedia, the signum (aka sign) function should return zero if zero

![image](https://user-images.githubusercontent.com/40372043/148807921-72f9eb5a-e079-46be-9256-a3de996d91ec.png)

There's also this answer from stackoverflow, which might be worth looking into as an optimization: https://stackoverflow.com/a/4609795/11622959

Reply:
I think the real question is "is there a distinct difference"? any mathematicians kicking around?

Reply:
> We could backport `SIGN` with the behavior of `0.0` returning `0.0` and keep `SGN` the way it is.

just to clarify: are you suggesting that I do this for my PR, as in add that Macro and utilize it to clean up the expression?
Or to do that as another PR, and clean up the expression with the assumption that the new macro will be available when necessary?